### PR TITLE
docs: reporting.Title best practices

### PR DESCRIPTION
**Pull request by fernflower**
Require Title to be generic for reports of the same
semantics.
Also some minor refactor of reporting section has been
done.

### CODE REVIEW

File: docs/source/actor-rhel7-to-rhel8.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ The main tools you will use for the actor development are listed below.
 
 ### leapp
 
-The leapp framework uses provides the libraries required to be imported by any actor and also a binary tool used to control the execution of actors within a workflow.  
+The leapp framework uses provides the libraries required to be imported by any actor and also a binary tool used to control the execution of actors within a workflow.
```
Comment (Rezney) [inline docs/source/actor-rhel7-to-rhel8.md]: I know this was not done but you but we can fix it here.

> uses provides the libraries

Comment (Rezney) [reply docs/source/actor-rhel7-to-rhel8.md]: apart from this looks good to me...

File: docs/source/actor-rhel7-to-rhel8.md [hunk 2]
```diff
@@ -161,20 +163,40 @@ class MyNewActor(Actor):
...
+create_report([
+    reporting.Title(
+-        'Upgrade process was interrupted because a deprecated module {0} is enabled'.format(module),
++        'Upgrade process was interrupted because a deprecated module is enabled',
+    reporting.Summary(
+        'Module {0} was surpassed by shiny-new-{0} and therefore it was '
+        'removed from RHEL-8. Keeping it in the configuration may '
+        'lock out the system thus it is necessary to disable it '
+        'before the upgrade process can continue.'.format(module)
+    ),
```
Comment (pirat89) [inline docs/source/actor-rhel7-to-rhel8.md]: I would go even further and skip generating of multiple reports for the same thing So summary would looks like:
```diff
-        'Module {0} was surpassed by shiny-new-{0} and therefore it was '
-        'removed from RHEL-8. Keeping it in the configuration may '
-        'lock out the system thus it is necessary to disable it '
-        'before the upgrade process can continue.'.format(module)
+        'Modules below have been surpassed by new modules and therefore these '
          ....
+        'before the upgrade process can continue. The list of affected modules:'
+        '    - modulA  -> Amod'
+        '    - modulB  -> Bmod'
```

Comment (pirat89) [reply docs/source/actor-rhel7-to-rhel8.md]: or it could start as well as 'We detected deprecated modules that are replaced....'

Comment (fernflower) [reply docs/source/actor-rhel7-to-rhel8.md]: I'd say we shouldn't restrict 1-type-1-report-entry-per-actor (imagine we have a remediation for each module, some are hints, some are commands, but they must be run in specific order), but I'll add your example as another way of fixing the trouble.

File: docs/source/actor-rhel7-to-rhel8.md (hunk 1 above)
Comment (fernflower) [reply docs/source/actor-rhel7-to-rhel8.md]: `The leapp framework provides the libraries` will do?

File: docs/source/actor-rhel7-to-rhel8.md (hunk 2 above)
Comment (pirat89) [reply docs/source/actor-rhel7-to-rhel8.md]: @fernflower But in this very specific case, it's just about names of modules. everythig else is same. I think we could later create list of examples, documenting several cases and explain why we choose to report them in this of another way.

File: docs/source/actor-rhel7-to-rhel8.md (hunk 1 above)
Comment (pirat89) [reply docs/source/actor-rhel7-to-rhel8.md]: I guess the original idea was: `The leapp framework uses the provided libraries ...`.  But @fernflower seems better to me.

Comment (Rezney) [reply docs/source/actor-rhel7-to-rhel8.md]: @fernflower sounds good...

**Pull request merged**